Notes/Algorithms/Data Structures/Hash-Tables.c

```diff
@@ -1,230 +1,244 @@
+    int deleteSuccess = deleteItem(testHash, key, value);
+    int searchSuccess = searchHashTable(testHash, key, value);
+    if (deleteSuccess == SUCCESS && searchSuccess == ERROR && expectedOutcome == SUCCESS) {
+        printf("The key/value pair: %s and %d was deleted successfully as expected. \n", key, value);
         return SUCCESS;
+    }  else if (deleteSuccess == ERROR && searchSuccess == ERROR && expectedOutcome == ERROR) {
+        printf("The key/value pair: %s and %d could not be deleted successfully as expected. \n", key, value);
+        return ERROR;
     } 
-    printf("The key %d could not be deleted successfully as expected. \n", key);
+    printf("Mismatch has occurred. Result does not match expected outcome for success or failure. \n");
     return ERROR;
 }
 
-
+// buildDeleteTests
+// creates 4 test cases expected to pass the testDelete function
+// creates 2 test cases (out of range) expected to fail the testDelete function
 void buildDeleteTests() {
     printf("Build Delete Test Passes: \n");
-    int testKeys1[10] = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 }; 
-    int testValues1[10] = { 0, 20, 30, 40, 50, 60, 70, 80 , 90, 100};
-    testDelete(1,testKeys1, testValues1);
+    testDelete("KeyOne", 10, SUCCESS);
+    testDelete("KeyTwo", 20, SUCCESS);
+    testDelete("KeyEight", 80, SUCCESS);
+    testDelete("KeyTen", 100, SUCCESS);
 
-    int testKeys2[10] = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 }; 
-    int testValues2[10] = {10, 20, 30, 40, 50, 60, 70, 0, 90, 100};
-    testDelete(8,testKeys2,testValues2);
 
-    int testKeys3[10] = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 }; 
-    int testValues3[10] = {10, 0, 30, 40, 50, 60, 70, 80 , 90, 100};
-    testDelete(2,testKeys3, testValues3);
-
-    int testKeys4[10] = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 }; 
-    int testValues4[10] = {10, 20, 30, 40, 0, 60, 70, 80 , 90, 100};
-    testDelete(5,testKeys4, testValues4);
-
-    int testKeys5[10] = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 }; 
-    int testValues5[10] = {10, 20, 0, 40, 50, 60, 70, 80 , 90, 100};
-    testDelete(3,testKeys5, testValues5);
-
-    //5 Fail
     printf("Build Delete Test Fails: \n");
-    int testKeys6[10] = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 }; 
-    int testValues6[10] = {10, 20, 30, 40, 50, 60, 70, 80 , 90, 100};
-    testDelete(-10,testKeys6, testValues6);
-
-    int testKeys7[10] = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 }; 
-    int testValues7[10] = {10, 20, 30, 40, 50, 60, 70, 80 , 90, 100};
-    testDelete(50,testKeys7, testValues7);
-  
+    // neither key nor value present
+    testDelete("Keyyyyyyy", 0, ERROR);
+    // key present value not
+    testDelete("KeyOne", 100, ERROR);
+    // key not present value present
+    testDelete("KeyZero", 100, ERROR);
 }
 
-void main() {
+//main calls buildSearchTests and buildDeleteTests in succession to begin the process of testing the search, reverse and delete functions
+void main() {    
    buildSearchTests();
    buildDeleteTests();
 }
```

Comment:
I'd like to see some test cases exercising the following behavior:

1. Collision logic (for small n, we can probably brute force a collision outside of the tests and then use that in them -- let me know if you want help with this)
2. Inserts > `hashTable->size`

Reply:
1: Added a testInsertCollision in [commit d52ed70682f92c86a7931567759b5ce0e21568d2](https://github.com/kcarrel/Training-Notes/pull/12/commits/d52ed70682f92c86a7931567759b5ce0e21568d2)
2: Added a testInsertCapacity in [commit c9cae280b193aa1b1e5a4a67c2719a900576c049](https://github.com/kcarrel/Training-Notes/pull/12/commits/c9cae280b193aa1b1e5a4a67c2719a900576c049)